README.md

```diff
@@ -32,12 +31,19 @@ jobs:
       - uses: actions/checkout@v2
       - uses: w3c/spec-prod@v2
         with:
-          GH_PAGES_BRANCH: gh-pages
+          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
+          # Replace following with appropriate value. See options.md for details.
+          W3C_WG_DECISION_URL: https://lists.w3.org/Archives/Public/public-group/2014JulSep/1234.html
 ```
 
+## More examples
 
-View all [**Available options** in docs](docs/options.md).
+Learn from [usage examples](docs/examples.md), including:
+ 
+ * [Run as a validator on pull requests](docs/examples.md#run-as-a-validator-on-pull-requests)
+ * [Selectively enable/disable validators](docs/examples.md#selectively-enabledisable-validators)
```

Comment:
Maybe replace this example with deploy to GitHub pages one

Reply:
Deploying to gh-pages is not a thing tho. Definitely not something we want to encourage. The only place to deploy is TR.

Reply:
Bikeshed needs GitHub pages deployment. Also, specs with large number of data-include or other slow processing should probably be encouraged to use static deploys.

Reply:
It doesn't *need* - it's something people do (but shouldn't). However, in the WebApps WG, for example, we deploy BS specs directly to TR.

Reply:
I agree we should encourage and facilitate WGs to deploy directly to TR.

But is in either-or? If a spec is being deployed directly to TR, does that mean we don’t want to also have it being published at a w3c.github.io URL? Is it bad practice to autopublish (autodeply) to both places? If so, why? What’s the harm that it causes?

Regardless, I think we need to consider that in practice there can be (or even, usually is) an interim period — before a WG has recorded a decision to autopublish to TR — when a WG wants to autopublish to GitHub pages first.

Existence proof: Currently none of the WebAppSec WG specs are being autopublished to TR — because there’s not been any decision from the group yet — or even CfC yet — to autopublish to TR. Yet, some of the the specs are getting updated regularly, and the editors and we want the EDs to automatically get updated on every push. So for now, I set up GitHub pages autopublishing for a number of those specs (using this tooling, spec-prod).

I will soon be working to get agreement from the group to autopublish to TR — but that takes some time, and it’s not something I can do unilaterally; it requires the chairs to send out that CfC, etc.

But even once we do get the WebAppSec specs set up to autopublish to TR, it’s not clear that me that necessarily means we should then stop using GitHub Pages and stop also autopublishing to w3c.github.io/webappsec-\* URLs. And it’s definitely not clear to me that’s what the editors or the members of the WG would want. Certainly it should be something that the members of the WG themselves are asked to decide on, and given a chance to decide on.

(Apologies in advance if I’ve misunderstood what’s being discussed/proposed here, and my comments are thus not actually relevant…)

Reply:
Definitely not setting any policy here, @sideshowbarker. What we are discussing is what should be in the primary example. It's basically for folks like you and me that actually do the grunt work of setting up the GitHub actions/workflows for working groups - and not something individual editors tend to do (in my experience, across all the groups I'm involved with).  

In the majority of cases, and given that this only happens every few months, I'd like to just land on the README, copy/pasta the auto-publish to TR, and be done (i.e., the 90% case - braindead easy, no reading docs, very little cognitive function). The case of "well, I want the ED, but like conditions X, Y, and Z." can live in the `examples/` if anyone wants to get fancy.

Reply:
> Definitely not setting any policy here, @sideshowbarker. What we are discussing is what should be in the primary example. It's basically for folks like you and me that actually do the grunt work of setting up the GitHub actions/workflows for working groups

OK yeah that makes sense.

> and not something individual editors tend to do (in my experience, across all the groups I'm involved with).

Yeah, that matches my experience too.

But I want to point out another apparent limitation in Echidna autopublishing to TR space that I’ve run into just today: If changes get pushed to a repo more than once a day, autopublishing of the first push will succeed — but it will fail for all subsequent pushes that day with the following error:

> "key": "same-this-and-previous",
> "detailMessage": "\"This\" version and \"previous\" version have the same URL.",

For a specific such failure, see https://github.com/w3c/webappsec-csp/runs/2101222121

And you know well that with some specs, it’s not at all uncommon for multiple changes to get pushed in one day.

So I don’t know any way to get around that. But maybe there’s some solution for it that I’m not aware of. I hope so.

But if not — if that’s a hard limitation in TR autopublishing (that we can’t autopublish a to a TR spec URL more than once a day), then it’d seem to make sense to not discourage editors from also publishing to GitHub Pages URLs. Otherwise, they have no way to ensure there’s a published version whose content matches the current state of commits in the repo.

Reply:
> So I don’t know any way to get around that. But maybe there’s some solution for it that I’m not aware of. I hope so.

I think if you set the previous published date to, say, the date of the FPWD, then I think it works.

Reply:
@sideshowbarker, for BikeShed specs, you can try: 
https://tabatkins.github.io/bikeshed/#metadata-previous-version

Reply:
> > So I don’t know any way to get around that. But maybe there’s some solution for it that I’m not aware of. I hope so.
> 
> I think if you set the previous published date to, say, the date of the FPWD, then I think it works.

Unless I’m understanding, I don’t think that actually works. Example:

https://www.w3.org/TR/CSP3/

Currently, that URL is essentially a symlink to https://www.w3.org/TR/2021/WD-CSP3-20210313/, and the Previous Versions link is to https://www.w3.org/TR/2021/WD-CSP3-20210310/

Now, if I decide I want to publish a new version for CSP in TR and I just keep the Previous Versions link set to https://www.w3.org/TR/2021/WD-CSP3-20210310/, then when Echidna runs, one of the things it does is that it checks the current contents at https://www.w3.org/TR/CSP3/ to see that they exactly match what we are telling Echidna we now want to make the Previous Version.

But if we keep Previous Version set to https://www.w3.org/TR/2021/WD-CSP3-20210310/ then that doesn’t match what’s at https://www.w3.org/TR/CSP3/ (which Echidna expects to be https://www.w3.org/TR/2021/WD-CSP3-20210313/

And so in that case, Echidna fails with a _“latest-is-not-previous”_ error:

https://labs.w3.org/echidna/api/status?id=a04327e5-f215-443a-8c65-6ddb300350ea

> "detailMessage": "Retrieved \"previous\" and \"latest\" documents, but their contents don't match.",

So with Bikeshed, the way to avoid that is to not put a hardcoded URL in the Bikeshed *Previous Version* metadata value — but to instead put the special value `from biblio`, which causes Bikeshed to look up the current Latest Version URL in its biblio DB, and use that for the Previous Version value.

And that all works perfectly — *except in the case where you push more than once a day*. Because in that case, you would keep trying to push to, for example, that `https://www.w3.org/TR/2021/WD-CSP3-20210313/` URL — that would become your Previous Version URL. But the problem is, that’s still also your This Version URL.

So Echidna will not let you do that. Instead it fails with that _“same-this-and-previous”_ (*“"This" version and "previous" version have the same URL.”*) message.

And like I said, as long as Echidna is doing that check, as far as I can see that’s gonna limit us to only being able to autopublish to that `https://www.w3.org/TR/2021/WD-CSP3-20210313/` URL exactly once. Echidna will refuse every subsequent attempt to autopublish to it again — because any attempt will either run into either the _“same-this-and-previous”_ failure or else the _“latest-is-not-previous”_ failure.

Reply:
> @sideshowbarker, for BikeShed specs, you can try "from":
> [tabatkins.github.io/bikeshed/#metadata-previous-version](https://tabatkins.github.io/bikeshed/#metadata-previous-version)

Yeah `from biblio` is what I’m already using — and it works beautifully except in the case where somebody pushes multiple changes in one day. That fails as described in https://github.com/w3c/spec-prod/pull/33#discussion_r595664294

Reply:
Serves them right for being too... (wait for it)... "pushy" 🥁🤣.